cmd/crash-dianostics.go

```diff
@@ -26,15 +26,19 @@ type globalFlags struct {
 func crashDiagnosticsCommand() *cobra.Command {
 	flags := &globalFlags{debug: false}
 	cmd := &cobra.Command{
-		Args:  cobra.NoArgs,
-		Use:   CliName,
-		Short: fmt.Sprintf("%s helps to troubleshoot kubernetes cluster", CliName),
-		Long:  fmt.Sprintf("%s collects diagnostics from an unresponsive Kubernetes cluster", CliName),
-		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			return preRun(flags)
-		},
+		Args:         cobra.NoArgs,
+		Use:          CliName,
+		Short:        fmt.Sprintf("%s helps to troubleshoot kubernetes cluster", CliName),
+		Long:         fmt.Sprintf("%s collects diagnostics from an unresponsive Kubernetes cluster", CliName),
 		SilenceUsage: true,
 		Version:      buildinfo.Version,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			err := CreateCrashdDir()
```

Comment:
Lets call `CreateCrashdDir` in `preRun` so that way your error handling is simpler.

Reply:
Move it here https://github.com/vmware-tanzu/crash-diagnostics/blob/9725e93afd82322bf742b55d32184c187b5d5862/cmd/crash-dianostics.go#L62